File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.js
function solution(input) {
    let answer = '';

    input.splice(1, input.length)
    .sort((a, b) => {
        if(a[0] === b[0]) {
            return a[1] - b[1];
        }
        return a[0] - b[0];
    })
    .forEach(el => {
        answer += `${el[0]} ${el[1]}\n`;
    })

    console.log(answer);
}

const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,    
});

let input = [];
let list = [];

rl.on("line", function (line) {
   input.push(line)
}).on("close", function () {
    input.forEach((val) => {
       list.push(val.split(' ').map(el => el));
    });
    solution(list);
    process.exit();
});

File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.py
"""
1. 입력 -> O(N): 100,000
2. 우선순위 큐 사용 -> SORT 혹은 FOR문 사용 시, 시간초과날 확률이 높음
2. 출력 -> 좌표 순서대로
"""
from collections import deque

deque = []
N = int(input())

for _ in range(N): # O(N)
    x, y = map(int, input().split())
    deque.append((x, y))

deque.sort(reverse=True)

for _ in range(N): # O(N)
    x, y = deque.pop()
    print(x, y)